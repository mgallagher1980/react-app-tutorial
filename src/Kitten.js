export const double =
    (numberToDouble) => numberToDouble * 2;

export const triple =
    (numberToTriple) => numberToTriple * 3;

const maths = {
    double: double,
    triple: triple
};

export default maths;


// class Kitten extends Component {
//   render() {
//     return (
//       <div>
//         KITTENS!
//       </div>
//     );
//   }
// }

// export default Kitten;
