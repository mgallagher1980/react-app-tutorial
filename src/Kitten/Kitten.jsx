import React, { Component, Fragment } from 'react';

export default class Kitten extends Component {
  render() {
    const kitten = this.props.kitten;
    const index = this.props.index;

    return (
      <Fragment>
        <span>{index + 1}</span>
      <p>{kitten.title}</p>
      <img
       src={kitten.images.preview_gif.url} 
       alt={kitten.title} />
       </Fragment>
    )
  }
}
