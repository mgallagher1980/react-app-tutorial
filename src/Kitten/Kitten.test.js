import Kitten from "./Kitten";
import React from "react";
import renderer from "react-test-renderer";
import ShallowRenderer from "react-test-renderer/shallow";
import { shallow } from "enzyme";

var kittenModel = {
  title: "something",
  images: {
    preview_gif: {
      url: "/"
    }
  }
};
describe('Kitten', () => {
  let kitten = null;

  beforeEach(() => {
    kitten = <Kitten kitten={kittenModel} />;
  });
  it('cuteness should be 99', () => {

    const rendered = renderer.create(kitten);

    const instance = rendered.getInstance();
    
    expect(instance.cuteness()).toBe(99);

  });

  it('should render the title as a paragraph', () => {

    const rendered = renderer.create(kitten);

    const p = rendered.root.findByType("p");

    expect(p.children.length).toBe(1);
    expect(p.children[0]).toBe(kittenModel.title);

  });

  it('should have title in a p', () => {
    const renderer = new ShallowRenderer();
    renderer.render(kitten);
    const result = renderer.getRenderOutput();

    expect(result.props.children[0].props.children).toBe(kittenModel.title);
  });

  it('should have title enzyme', () => {
    const rendered = shallow(kitten);

    expect(rendered.find("p").text()).toBe(kittenModel.title);
  });

  it('should match snapshot2', () => {
    const rendered = renderer.create(kitten);
    const tree = rendered.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
});