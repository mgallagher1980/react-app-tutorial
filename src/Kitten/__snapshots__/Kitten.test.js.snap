// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Kitten should match snapshot2 1`] = `
Array [
  <p>
    something
  </p>,
  <img
    alt="something"
    src="/"
  />,
]
`;
